src/utils/Router.js

```diff
@@ -0,0 +1,36 @@
+import { Router } from 'express';
+import permissionsValidator from '../validators';
+
+const methods = ['post', 'get', 'put', 'delete'];
+
+class ExtendRouter {
+  constructor() {
+    let router = Router();
+
+    router = this.setMethods(router);
+
+    return router;
+  }
+
+  setMethods(router) {
+    const modifyRouter = router;
+    methods.forEach((method) => {
+      modifyRouter[method] = this.setMethod(method);
+    });
+    return modifyRouter;
+  }
+
+  setMethod = (method) => {
+    function setMethod(path, permissionLevel, ...args) {
```

Comment:
why do you need a closure here?))

There's no need to be super generic here with looping and what not.

It's much better and more understandable to actually declare 4 methods (it's not hard, I promise).
I know it's cool to automate things, but there's only so mane methods you will ever need, this code will not change too much and we're not expecting methods array to be 100+ entries long.

so doing something along the lines of 
{
  get = (route, permission, controllerMethod) => {
     this.router.get(route, permissionsValidator(permissionLevel));
     this.router.get(route, controllerMethod);
  }
}

should be fine for now (you might have to deal with context loss, but bind and apply should help with that);

Don't overcomplicate things with closures and unnecessary automation